Tests/BugsnagClientTests.m

```diff
@@ -218,4 +222,49 @@ - (void)testChangesToConfigurationAreIgnoredAfterCallingStart {
     [self assertEqualConfiguration:initialConfig withActual:configAfter];
 }
 
+
+/**
+ * Verifies that a large breadcrumb is not dropped (historically there was a 4kB limit)
+ */
+- (void)testLargeBreadcrumbSize {
+    BugsnagConfiguration *configuration = [[BugsnagConfiguration alloc] initWithApiKey:DUMMY_APIKEY_32CHAR_1];
+    configuration.enabledBreadcrumbTypes = BSGEnabledBreadcrumbTypeNone;
+    BugsnagClient *client = [[BugsnagClient alloc] initWithConfiguration:configuration];
+    [client start];
+
+    NSMutableArray *breadcrumbs = client.configuration.breadcrumbs.breadcrumbs;
+    XCTAssertEqual(0, [breadcrumbs count]);
+
+    // small breadcrumb can be left without issue
+    [client leaveBreadcrumbWithMessage:@"Hello World"];
+    XCTAssertEqual(1, [breadcrumbs count]);
+
+    // large breadcrumb is also left without issue
+    __block NSUInteger crumbSize = 0;
+
+    [client addOnBreadcrumbBlock:^BOOL(BugsnagBreadcrumb *breadcrumb) {
+        NSData *data = [NSJSONSerialization dataWithJSONObject:[breadcrumb objectValue] options:0 error:nil];
+        crumbSize = data.length;
+        return true;
+    }];
+
+    NSDictionary *largeMetadata = [self generateLargeMetadata];
+    [client leaveBreadcrumbWithMessage:@"Hello World"
+                              metadata:largeMetadata
+                               andType:BSGBreadcrumbTypeManual];
+    XCTAssertTrue(crumbSize > 4096); // previous 4kb limit
+    XCTAssertEqual(2, [breadcrumbs count]);
```

Comment:
Is it possible to check that the breadcrumb message/metadata hasn't been truncated, despite a breadcrumb entry having been successfully added?

Reply:
Updated to verify that the metadata + message are equal to the original values after a breadcrumb has been recorded by the Client.